src/workspace-data/useDataTableProvider.ts

```diff
@@ -33,98 +37,122 @@ export const useDataTableProvider = (
         });
-    },
-    [signal, setWdsTypes]
-  );
+    }
+  }, [signal, setWdsTypes, wdsUrl, workspaceId]);
 
-  const loadWdsCapabilities = useCallback(
-    async (url) => {
+  const loadWdsCapabilities = useCallback(async () => {
+    if (wdsUrl.status === 'Ready') {
       try {
-        const capabilitiesResult = await Ajax(signal).WorkspaceData.getCapabilities(url);
+        const capabilitiesResult = await Ajax(signal).WorkspaceData.getCapabilities(wdsUrl.state);
         setWdsCapabilities({ status: 'Ready', state: capabilitiesResult });
       } catch (error) {
         setWdsCapabilities({ status: 'Error', state: null, error: 'Error loading WDS capabilities' });
         reportError('Error loading WDS capabilities', error);
       }
-    },
-    [signal, setWdsCapabilities]
-  );
+    }
+  }, [signal, setWdsCapabilities, wdsUrl]);
 
   const loadWdsData = useCallback(async () => {
-    // Try to load the proxy URL
-    if (!wdsApp || !['Ready', 'Error'].includes(wdsApp.status)) {
-      const foundApp = await loadWdsApp(workspaceId);
-      // TODO: figure out how not to make this redundant fetch, per
-      // https://github.com/DataBiosphere/terra-ui/pull/4202#discussion_r1319145491
-      if (foundApp) {
-        loadWdsTypes(foundApp.proxyUrls?.wds, workspaceId);
-        loadWdsCapabilities(foundApp.proxyUrls?.wds);
+    if (useCwds.status === 'Ready') {
+      // Try to load the proxy URL
+      if (!['Ready', 'Error'].includes(wdsApp.status)) {
+        await loadWdsApp();
       }
     }
-    // If we have the proxy URL try to load the WDS types
-    else if (wdsApp?.status === 'Ready') {
-      const proxyUrl = wdsApp.state.proxyUrls?.wds;
-      await loadWdsTypes(proxyUrl, workspaceId);
-      await loadWdsCapabilities(proxyUrl);
+  }, [wdsApp, loadWdsApp, useCwds]);
+
+  useEffect(() => {
+    if (useCwds.status === 'Ready' && wdsApp.status === 'None') {
+      loadWdsApp();
+    }
+    if (wdsUrl.status === 'Ready') {
+      loadWdsTypes();
+      loadWdsCapabilities();
+    }
+  }, [wdsUrl, loadWdsTypes, loadWdsCapabilities, useCwds, wdsApp, loadWdsApp]);
+
+  useEffect(() => {
+    const checkCWDS = async (): Promise<void> => {
+      try {
+        const response = await fetchWDS(cwdsURL)(`collections/v1/${workspaceId}`, _.merge(authOpts(), { signal }));
+        const data = await response.json();
+        if (Object.keys(data).length !== 0) {
+          setWdsUrl({ status: 'Ready', state: cwdsURL });
+          setUseCwds({ status: 'Ready', state: true });
+          // setWdsApp({ status: 'Ready', state: undefined }); // No app needed for CWDS
+        } else {
+          setWdsUrl({ status: 'Loading', state: null });
+          setUseCwds({ status: 'Ready', state: false });
+        }
+      } catch (error) {
+        console.error(error);
+        setWdsUrl({ status: 'Loading', state: null });
+        setUseCwds({ status: 'Ready', state: false });
+      }
+    };
```

Comment:
```suggestion
    const checkCWDS = async (): Promise<void> => {
      // set default status
      setWdsUrl({ status: 'Loading', state: null });
      setUseCwds({ status: 'Ready', state: false });
      try {
        const response = await fetchWDS(cwdsURL)(`collections/v1/${workspaceId}`, _.merge(authOpts(), { signal }));
        if (response.status == 200) {
          setWdsUrl({ status: 'Ready', state: cwdsURL });
          setUseCwds({ status: 'Ready', state: true });
        } 
      } catch (error) {
        console.error(error);
      }
    };
```

Reply:
I don't want to set `useCwds` to false until I know that it's actually false, otherwise we'll end up looking for an app when there shouldn't be one.  Likewise, I expect the response status to be 200 even if no collections exist in the workspace, it would just be an empty response.